code/modules/spells/ability_types/realized_aimed.dm

```diff
@@ -415,3 +415,124 @@
 		L.apply_damage(ishuman(L) ? (damage_amount - distance_decrease)*0.5 : (damage_amount - distance_decrease), BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
 		var/atom/throw_target = get_edge_target_turf(L, get_dir(L, get_step_towards(L, get_turf(src))))
 		L.throw_at(throw_target, 1, 2)
+
+/obj/effect/proc_holder/ability/aimed/yin_laser
+	name = "Anarchy"
+	desc = "An ability that summons a devastating laser. \
+		If another player gets hit by the laser they get +20 justice for a period of time. \
+		If the person that gets hit is wearing duality of harmony, they will get huge buffs to their defenses and stats."
+	action_icon_state = "yinform0"
+	base_icon_state = "yinform"
+	cooldown_time = 60 SECONDS
+	var/laser_range = 20
+	var/list/spawned_effects = list()
+
+/obj/effect/proc_holder/ability/aimed/yin_laser/Perform(target, user)
+	var/turf/t_turf = get_turf(target)
+	INVOKE_ASYNC(src, .proc/Cast, t_turf, user)
+	return ..()
+
+/obj/effect/proc_holder/ability/aimed/yin_laser/proc/Cast(turf/target, mob/user)
+	user.face_atom(target)
+	var/mob/living/carbon/human/H = user
+	var/turf/my_turf = get_turf(H)
+	var/turf/target_turf = get_ranged_target_turf_direct(user, target, laser_range)
+	var/list/to_hit = getline(user, target_turf)
+	var/datum/beam/beam  = my_turf.Beam(target_turf, "volt_ray")
+	for(var/turf/open/OT in to_hit)
+		if(!istype(OT) || OT.density)
+			break
+		beam.target = OT
+		beam.redrawing()
+		sleep(1)
+		new /obj/effect/temp_visual/revenant/cracks/yinfriend(OT)
+	for(var/obj/effect/FX in spawned_effects)
+		qdel(FX)
+	qdel(beam)
+
+/obj/effect/temp_visual/revenant/cracks/yinfriend
+	icon_state = "yincracks"
+	duration = 9
+	var/damage = 500  // Amount of black damage dealt to enemies from the laser.
+	var/list/faction = list("neutral")
+
+/obj/effect/temp_visual/revenant/cracks/yinfriend/Destroy()
+	for(var/turf/T in range(1, src))
+		if(T.z != z)
+			continue
+		for(var/mob/living/L in T)
+			if(faction_check(L.faction, src.faction))
+				continue
+			L.apply_damage(damage, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+		for(var/mob/living/carbon/human/L in T)
+			if(!faction_check(L.faction, src.faction))
+				continue
+			if(L.status_flags & GODMODE)
+				continue
+			if(L.stat == DEAD)
+				continue
+			if(L.is_working) //no work heal :(
+				continue
```

Comment:
```suggestion
```
None of these checks are particularly relevant to any scenario that this buff would be disruptive to.